src/contexts/ProductContext.js

```diff
@@ -0,0 +1,125 @@
+import axios from "axios";
+import { createContext, useContext, useEffect, useState ,useReducer } from "react";
+
+const productList = createContext();
+
+const ProductListProvider = ({children  }) => {
+
+    const [product, setProduct ] = useState([]);
+
+    useEffect(()=>{
+        (async () => {
+            try{const response = await axios.get("/api/products");
+            setProduct(response.data.products)
+        }
+        catch (error){
+            console.log(error)
+        }
+        })();
+    },[]);
+
+    const FilterReducer = (state,action) =>{
```

Comment:
Suggestion: You can create a separate folder for all your reducer implementations.